exercises/practice/pythagorean-triplet/.meta/example.sml

```diff
@@ -0,0 +1,21 @@
+(* a*a + b*b = c*c, a + b + c = n
+ * a*a + b*b = (n - a - b) * (n - a - b)
+ * 0 = n*n - 2*n*a - 2*n*b + 2*a*b
+ * (2*n - 2*a) b = (n*n - 2*n*a)
+ *)
```

Comment:
The docstring is already good but I think you could improve it by finally pointing out: "Hence every triplet has to satisfy $b=\frac{n^2 - 2na}{2n-2a}$" (even better if the reverse implication also holds).

In line 3 and 4 of your derivation you could prepend `<=> ` to make it clearer that you perform equivalence transformations. For the first transformation it would be helpful if you just say that you substitute `c ` (this also helps to gain confidence that the formula for `b` also implies that `a`, `b`, `c` is a pythagorean triplet).

Reply:
Comment updated.

Reply:
I wouldn't use the latex formula for `b`, it does not match the style of the rest of the docstring. You can just write `<=> b = n*(n - 2*a) / (2*(n - a))`.

Reply:
Updated